src/cdk/testing/testbed/fake-events/type-in-element.ts

```diff
@@ -60,20 +80,31 @@ export function typeInElement(element: HTMLElement, ...modifiersAndKeys: any) {
   for (const key of keys) {
     dispatchKeyboardEvent(element, 'keydown', key.keyCode, key.key, element, modifiers);
     dispatchKeyboardEvent(element, 'keypress', key.keyCode, key.key, element, modifiers);
-    if (isTextInput(element) && key.key && key.key.length === 1) {
-      element.value += key.key;
+    if (isInputAware(element) && key.key?.length === 1) {
+      if (isTextInput(element)) {
+        element.value += key.key;
+      } else {
+        element.appendChild(new Text(key.key));
```

Comment:
What is the difference between appending text and setting the `textContent` like you're doing in the `clear` method? Would it be easier to just always set `textContent` and extract a common `setText(element, text)` method to do that?

Reply:
`contenteditable` element may contain rich content, aka non-text elements, which we shouldn't change.